bin/dfx-ckbtc-import

```diff
@@ -0,0 +1,131 @@
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+. "$SOURCE_DIR/versions.bash"
+
+print_help() {
+  cat <<-EOF
+
+	Adds ckbtc canisters to the local dfx.json and populates the corresponding wasm and did files.
+
+	EOF
+}
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=c long=commit desc="Commit of the IC repo to download canisters from" variable=DFX_IC_COMMIT default="$DFX_IC_COMMIT"
+clap.define short=p long=prefix desc="Prefix for the local canister names" variable=LOCAL_PREFIX default="ckbtc_"
+clap.define long=check desc="Check whether the canisters have been imported" variable=DFX_CHECK nargs=0
+clap.define long=wasm_dir desc="Where to store the imported wasm files" variable=WASM_DIR default="wasms"        # Matches `dfx nns import`
+clap.define long=candid_dir desc="Where to store the imported candid files" variable=CANDID_DIR default="candid" # Matches `dfx nns import`
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+# Imports wasm, candid and populate dfx.json
+ckbtc_import() {
+  test -e dfx.json || {
+    echo "ERROR: No dfx.json found.  Please run this from inside a dfx project."
+    exit 1
+  } >&2
+
+  mkdir -p "$WASM_DIR"
+  mkdir -p "$CANDID_DIR"
+
+  : Get the wasms
+  get_wasm() {
+    local remote_name="$1"
+    local local_name="$2"
+    if test -e "$local_name"; then
+      echo "Skipping $local_name as it already exists"
+    else
+      local url="https://download.dfinity.systems/ic/$DFX_IC_COMMIT/canisters/${remote_name}.gz"
+      echo "Getting  $local_name from $url..."
+      curl -sSL "$url" -o "${local_name}.gz"
+      gunzip "${local_name}.gz"
+    fi
+  }
+  pushd "$WASM_DIR"
+  get_wasm ic-ckbtc-minter.wasm "${LOCAL_PREFIX}minter.wasm"
+  get_wasm ic-icrc1-ledger.wasm "${LOCAL_PREFIX}ledger.wasm"
+  get_wasm ic-icrc1-index.wasm "${LOCAL_PREFIX}index.wasm"
+  popd
+
+  : Get the candid files
+  get_did() {
+    local remote_name="$1"
+    local local_name="$2"
+    if test -e "$local_name"; then
+      echo "Skipping $local_name as it already exists"
+    else
+      local url="https://raw.githubusercontent.com/dfinity/ic/$DFX_IC_COMMIT/${remote_name}"
+      echo "Getting  $local_name from $url..."
+      curl -sSLf --retry 5 "$url" -o "$local_name"
+    fi
+  }
+  pushd "$CANDID_DIR"
+  get_did rs/bitcoin/ckbtc/minter/ckbtc_minter.did "${LOCAL_PREFIX}minter.did"
+  get_did rs/rosetta-api/icrc1/ledger/icrc1.did "${LOCAL_PREFIX}ledger.did"
+  get_did rs/rosetta-api/icrc1/index/index.did "${LOCAL_PREFIX}index.did"
+  popd
+
+  : Update dfx.json
+  LOCAL_PREFIX="${LOCAL_PREFIX:-}" \
+    jq ' .canisters["\(env.LOCAL_PREFIX)ledger"] = { build: "true", candid: "candid/\(env.LOCAL_PREFIX)ledger.did", wasm: "wasms/\(env.LOCAL_PREFIX)ledger.wasm", type: "custom", remote: { id: {mainnet: "mxzaz-hqaaa-aaaar-qaada-cai" } }}
```

Comment:
Does this change dfx.json?

Reply:
Yes, it adds canister entries to dfx.json, like `dfx nns install` but unlike the download script.  Is that OK for your use cases?

Reply:
I'd rather have them in dfx.json directly. We already have three entries. We either use them or remove them.

Reply:
Ok.  I'll make the addition optional, then.  This way the code is consistent with `dfx nns install` but doesn't change dfx.json if that is unwanted.

Reply:
thanks